apps/web/src/hooks/useThemeChange.tsx

```diff
@@ -0,0 +1,42 @@
+import { ColorScheme, useMantineColorScheme } from '@mantine/core';
+import { IconDarkMode, IconLightMode, IconTonality } from '@novu/design-system';
+import { ColorSchemePreferenceEnum, THEME_TITLE_LOOKUP, useLocalThemePreference, useSegment } from '@novu/shared-web';
+import { useEffect, useMemo } from 'react';
+import { useDebounce } from './useDebounce';
+
+type ThemeChange = { colorScheme: ColorScheme; themeStatus: string };
+
+const getThemeIcon = (themeStatus: ColorSchemePreferenceEnum) => {
+  if (themeStatus === ColorSchemePreferenceEnum.DARK) {
+    return <IconDarkMode />;
+  }
+  if (themeStatus === ColorSchemePreferenceEnum.LIGHT) {
+    return <IconLightMode />;
+  }
+
+  return <IconTonality />;
+};
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** This whole thing would also fit best IMO in `design-system` because if our other apps wanted to use the theme-based icons, they'd have to re-implement this‏

Reply:
We primarily use the `useMantineColorScheme` as it relates more to the theme.

``` js
  const { colorScheme } = useMantineTheme();
  const isDark = colorScheme === 'dark';
```
It is used for both component designs and icons.
On second thought, I think this can become its own hook, `useIsDarkTheme.`

```js
export function useIsDarkTheme() {
  const { colorScheme } = useMantineTheme();
  const isDark = colorScheme === 'dark';

  return isDark;
}
```

Reply:
Indeed, the `theme` abstraction still belongs at the DS level IMO!

Re: `useIsDarkTheme`: It could, but we won't be needing it in the next iteration of the design system!

Reply:
Do you mean to move the functionality of getting the `IconDarkMode` `IconLightMode` etc to DS?